pkg/utils/yaml.go

```diff
@@ -102,6 +102,11 @@ func ScanIacDocumentsFromYaml(scanner *bufio.Scanner, byteArray []byte, filePath
 
 	dec := yaml.NewDecoder(bytes.NewReader(byteArray))
 
+	if dec == nil {
```

Comment:
Though the nil check is good, looking at the yaml.NewDecoder method, it will either panic or return a non nil decoder.